vpr/src/device/rr_graph_view.h

```diff
@@ -96,6 +96,60 @@ class RRGraphView {
         return node_storage_.fan_in(node);
     }
 
+    /* Get the xlow of a routing resource node. This function is inlined for runtime optimization. */
+    inline short node_xlow(RRNodeId node) const {
+        return node_storage_.node_xlow(node);
+    }
+
+    /* Get the xhigh of a routing resource node. This function is inlined for runtime optimization. */
+    inline short node_xhigh(RRNodeId node) const {
+        return node_storage_.node_xhigh(node);
+    }
+
+    /* Get the ylow of a routing resource node. This function is inlined for runtime optimization. */
+    inline short node_ylow(RRNodeId node) const {
+        return node_storage_.node_ylow(node);
+    }
+
+    /* Get the yhigh of a routing resource node. This function is inlined for runtime optimization. */
+    inline short node_yhigh(RRNodeId node) const {
+        return node_storage_.node_yhigh(node);
+    }
+
+    /* Get the length of a routing resource node. This function is inlined for runtime optimization. */
```

Comment:
I think the ``node_length()`` API may need a detailed comment. 
- The node length is only applicable to nodes that represent routing tracks. So you need to add
   ``VTR_ASSERT(node_type() == CHANX  || node_type() == CHANY);``
  This also has to be commented so that developers know when to use the API.
- We need to ensure if the ``node_length()`` can be negative or not. Can you quickly check the use of ``set_node_xhigh()``,  ``set_node_yhigh()``, ``set_node_xlow()`` and ``set_node_ylow()``, to see if the ``xhigh/yhigh`` is always larger than ``xlow/ylow``.  I know this is definitely true for any ``INC`` directional wires. But I forgot if the ``DEC`` wires apply the same rule or not. If so, we may consider to add a comment ``VTR_ASSERT_SAFE`` to ensure that the node_length() is always valid. You may also state in the comment that the node_length() is always a positive number.
- I see some codes use the ``node_length() - 1``. For me, the current ``node_length()`` definition is o.k. which shows ``1`` for a length-1 wire. Please double check the definition and see if we need more API or not.
- You may consider to add a figure in ASICII format to illustrate the idea.